classes/Module/ModuleRepository.php

```diff
@@ -0,0 +1,152 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License (AFL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0  Academic Free License (AFL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\AutoUpgrade\Module;
+
+use PrestaShop\Module\AutoUpgrade\AutoupgradeException;
+
+/**
+ * Class ModuleRepository is used to get lists of modules (those on the disk, native modules by version, ...)
+ */
+class ModuleRepository
+{
+    /** @var string */
+    private $modulesDir;
+
+    /** @var string */
+    private $disabledModulesDir;
+
+    /** @var AddonsClientInterface */
+    private $addonsClient;
+
+    /** @var array */
+    private $modulesByVersion;
+
+    /**
+     * @param string $modulesDir
+     * @param string $disabledModulesDir
+     * @param AddonsClientInterface $addonsClient
+     */
+    public function __construct(
+        $modulesDir,
+        $disabledModulesDir,
+        AddonsClientInterface $addonsClient
+    ) {
+        $this->modulesDir = rtrim($modulesDir, DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR;
```

Comment:
Better to not use `DIRECTORY_SEPARATOR` constant in PHP.
If you use DIRECTORY_SEPARATOR, which resolves to \ on windows, things tend to break. For example:\ gets converted to %5C in urls.